person.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Represent a super class from where teacher and student classes inherit
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(name = 'unknown', age = nil, parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
+
+  public
+
+  def can_use_services?
+    return true if @age >= 18 || @parent_permission == true
+
+    false
+  end
```

Comment:
[OPTIONAL] I suggest you move the `can_use_services?` method to come just after the constructor and that way there won't be any need to declare it as a public method this looks more professional and saves you some lines of code😉.

```suggestion
  def can_use_services?
    return true if @age >= 18 || @parent_permission == true

    false
  end

  private

  def of_age?
    return true if @age >= 18

    false
  end
```